### PR TITLE
Aria: Fix buffers overflow, remove unnecessary writebacks

### PR DESCRIPTION
**Pull request by pblaszko**
Commit 1:
Limiting Aria processing size to IBS. It fixes Aria internal buffers overflow.

Commit 2:
Removing unnecessary writebacks of internal Aria buffers

### CODE REVIEW

File: src/audio/aria/aria.c [hunk 1]
```diff
@@ -318,29 +317,26 @@ static int aria_copy(struct comp_dev *dev)
 	sink_c = buffer_acquire(sink);
 
 	comp_get_copy_limits(source_c, sink_c, &c);
-	source_bytes = c.frames * c.source_frame_bytes;
-	sink_bytes = c.frames * c.sink_frame_bytes;
+	copy_bytes = MIN(c.frames * c.source_frame_bytes, cd->base.ibs);
+	copy_samples = copy_bytes / sizeof(uint32_t);
```
Comment (btian1) [inline src/audio/aria/aria.c:321]: fixed with int32? not possible for int16.
if fixed, then simply >> 2.

**Review by tmleman — APPROVED** [review]
LGTM and fix issue that I have found playing with pipeline period.

**Review by dnikodem — APPROVED** [review]

**Review by kv2019i — APPROVED** [review]

**Comment by kv2019i** [comment]
SOFCI TEST

File: src/audio/aria/aria.c [hunk 2]
```diff
@@ -318,29 +317,28 @@ static int aria_copy(struct comp_dev *dev)
 	sink_c = buffer_acquire(sink);
 
 	comp_get_copy_limits(source_c, sink_c, &c);
-	source_bytes = c.frames * c.source_frame_bytes;
-	sink_bytes = c.frames * c.sink_frame_bytes;
+	copy_bytes = MIN(c.frames * c.source_frame_bytes, cd->base.ibs);
```
Comment (lyakh) [inline src/audio/aria/aria.c:320]: you don't need to take sink frame bytes into account at all?

Comment (pblaszko) [reply src/audio/aria/aria.c:320]: Aria supports only 32bit containers. Sink frame bytes is always equal source frame bytes.

**Review by marcinszkudlinski — APPROVED** [review]

**Pull request merged**